core/chains/evm/logpoller/log_poller.go

```diff
@@ -497,10 +534,30 @@ func (lp *logPoller) getCurrentBlockMaybeHandleReorg(ctx context.Context, curren
 	return currentBlock, nil
 }
 
+// backfillFinalizedBlocks backfills finalized blocks if we can for performance. If we crash during backfill, we
+// may reprocess logs.  Log insertion is idempotent so this is ok.
+// E.g. 1<-2<-3(currentBlockNumber)<-4<-5<-6<-7(latestBlockNumber), finality is 2. So 3,4 can be batched.
+// Although 5 is finalized, we still need to save it to the db for reorg detection if 6 is a reorg.
+// start = currentBlockNumber = 3, end = latestBlockNumber - finality - 1 = 7-2-1 = 4 (inclusive range).
+func (lp *logPoller) backfillFinalizedBlocks(ctx context.Context, currentBlockNumber *int64, latestBlockNumber int64) (ok bool) {
```

Comment:
more readable to make this signature:

```backfillFinalizedBlocks(ctx context.Context, currentBlockNumber int64, latestBlockNumber int64) (int64, error)```

then just return ```0, err``` for the error case. No mutating of input args and no nil pointer risk